src/functions.cpp

```diff
@@ -1,6 +1,7 @@
 #include "../inc/functions.hpp"    // TODO: add proper compilation flag and change to #include "functions.hpp"
 #include <iostream>
 #include <limits>
+#include <algorithm>
```

Comment:
You have proper flag (-Iinc/), so as Kasperice says, you can add headers straight by name.
Second case: sort headers. ;)